test-harness.sh

```diff
@@ -0,0 +1,40 @@
+#!/usr/bin/env bash
+
+START=$(date "+%s")
+
+set -e
+
+ENV_FILE=".test-env"
+source $ENV_FILE
+
+rootdir=$(dirname "$0")
+pushd "$rootdir"
+
+## Reset test harness
+if [ -d "$SDK_TESTING_HARNESS" ]; then
+  pushd "$SDK_TESTING_HARNESS"
+  ./scripts/down.sh
+  popd
+  rm -rf "$SDK_TESTING_HARNESS"
+else
+  echo "test-harness.sh: directory $SDK_TESTING_HARNESS does not exist - NOOP"
+fi
+
+git clone --single-branch --branch "$SDK_TESTING_BRANCH" "$SDK_TESTING_URL" "$SDK_TESTING_HARNESS"
+
+## OVERWRITE incoming .env with .test-env
+cp "$ENV_FILE" "$SDK_TESTING_HARNESS"/.env
```

Comment:
I've been grappling with this same issue. Here are some pros/cons for _keeping_ `.env` as is [in the SDK Testing Repo:](https://github.com/algorand/algorand-sdk-testing/blob/7af3ace6c9e66de45c20e5067fa4a63d8df8582a/.env)
* Pros
  1. Communicates and sets the standard defaults required to bootstrap the test harness. Any "future" SDK that wants to incorporate testing, has a reasonable starting point.
  2. `up.sh` in SDK testing doesn't need to know about other environment variables that need sourcing except for the standard `.env`
  3. By keeping all configurations in a single file `.test-env` we know exactly where configurations should be provided
* Cons
  1. Can achieve the same (as pro # 1 above) with the former `.up-env` (or copying ast `.test-env`) that doesn't overwrite SDK testing's `.env` and get similar effect by sourcing in sequence (however pro # 2 gives a counter-argument)
  2. We can make SDK Testing's `.env` empty and communicate to the downloader that they need to provide the variables

After writing these pros/cons down, I've mostly convinced myself that SDK Testing _should_ have an env. But I'm intrigued by the idea of renaming `.test-env` as `.env` to gain some consistency. The down side (or maybe up side) is that we then need to consider the impact on the python `Dockerfile` since `.env` is read by the docker build, unless explicitly ignored in a `.dockerignore`

Reply:
But I could also be missing some pros/cons. I welcome additions.

Reply:
> up.sh in SDK testing doesn't need to know about other environment variables that need sourcing except for the standard .env

This seems to be a good reason to me - maybe we can consider pruning or even deleting the `.env` file in the SDK Testing repo then.

Reply:
The pros/cons are a bit confusing and too much double negation is going on. My argument for `up.sh` not needing to know about other env vars means that `up.sh` need only call `source .env`. 

However, not providing an actual `.env` in SDK Repo in that case, means that the script would be broken as a standalone. I think the scripts in the SDK Testing repo should be able to run by themselves, without another repo needing to tweak them for them to work properly. That's the ultimate argument for having a `.env` in that repo, and for having the `.env` contain the minimum necessary env vars.